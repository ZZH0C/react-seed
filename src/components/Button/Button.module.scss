.button {
  color: white;
  border: 1px solid currentColor;
  border-radius: 3px;
  padding: 1rem 1.5rem;
  cursor: pointer;
}

.primary {
  background-color: #1ea7fd;

  &[disabled] {
    background-color: #e0f5fc;
  }
}

.secondary {
  background-color: #2f4f4f;

  &[disabled] {
    background-color: #606969;
  }
}
