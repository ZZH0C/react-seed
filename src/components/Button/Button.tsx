import React from 'react';
import classNames from 'classnames';
import styles from './Button.module.scss';

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  styleType: 'primary' | 'secondary';
}

export const Button: React.FC<ButtonProps> = ({
  className,
  styleType,
  ...props
}) => {
  return (
    <button
      className={classNames(className, styles.button, styles[styleType])}
      {...props}
    />
  );
};
