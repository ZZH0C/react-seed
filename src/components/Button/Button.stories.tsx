import { Button } from './Button';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

export default {
  title: 'UI/Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  styleType: 'primary',
  children: 'Button',
  disabled: false,
  onClick: () => alert('Button clicked!'),
};
