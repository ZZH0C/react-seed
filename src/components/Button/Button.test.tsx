import { shallow } from 'enzyme';
import { Button } from './Button';

describe('components/Button', () => {
  it('should have primary style render', () => {
    expect(shallow(<Button styleType="primary" />)).toMatchSnapshot();
  });

  it('should have secondary style render', () => {
    expect(shallow(<Button styleType="secondary" />)).toMatchSnapshot();
  });
});
