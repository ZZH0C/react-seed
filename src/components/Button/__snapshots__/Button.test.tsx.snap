// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button should have primary style render 1`] = `
<button
  className="button primary"
/>
`;

exports[`components/Button should have secondary style render 1`] = `
<button
  className="button secondary"
/>
`;
